framework/db/Connection.php

```diff
@@ -607,6 +611,7 @@ protected function initConnection()
         if ($this->charset !== null && in_array($this->getDriverName(), ['pgsql', 'mysql', 'mysqli', 'cubrid'])) {
             $this->pdo->exec('SET NAMES ' . $this->pdo->quote($this->charset));
         }
+        $this->getSupportsResultSets();
```

Comment:
OK. With last sentence you mean caching in cache backend? I think `prepare()` itself will not contact db at all (maybe it will with emulatePrepare = false?). Same with `nextRowSet()` I guess (`execute()` was not called at that point, so there can be no row set). I will figure this out.